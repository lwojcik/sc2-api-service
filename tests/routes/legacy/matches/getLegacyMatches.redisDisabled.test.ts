import fastify from 'fastify';
import server from '../../../../src/index';
import getConfig from '../../../helper';

describe('/legacy/matches/:regionId/:realmId/:profileId (Redis disabled)', () => {
  const fastifyServer = fastify() as any;
  const url = '/legacy/matches/1/1/1';

  beforeAll(() => fastifyServer.register(server, getConfig(false)));

  afterAll(() => fastifyServer.close());

  it('returns 200', async () => {
    expect.assertions(1);
    const res = await fastifyServer.inject({ method: 'GET', url });
    expect(res.statusCode).toBe(200);
  });

  it('returns correct response', async () => {
    expect.assertions(1);
    const res = await fastifyServer.inject({ method: 'GET', url });
    expect(res.payload).toMatchSnapshot();
  });
});
