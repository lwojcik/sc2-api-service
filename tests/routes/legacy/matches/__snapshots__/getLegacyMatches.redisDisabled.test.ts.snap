// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/legacy/matches/:regionId/:realmId/:profileId (Redis disabled) returns correct response 1`] = `"{\\"status\\":200,\\"data\\":{\\"matches\\":[{\\"map\\":\\"sample legacy match history object\\",\\"date\\":\\"\\"}]}}"`;
